pyccel/ast/basic.py

```diff
@@ -9,11 +9,15 @@
 - Basic, which provides a python AST
 - PyccelAstNode which describes each PyccelAstNode
 """
-
+import ast
 from sympy.core.basic import Basic as sp_Basic
 
 __all__ = ('Basic', 'PyccelAstNode')
 
+dict_keys   = {}.keys().__class__
+dict_values = {}.values().__class__
+iterable_types = (list, tuple, dict_keys, dict_values)
+
 #==============================================================================
 class Basic(sp_Basic):
```

Comment:
I think that you should provide a few dedicated unit tests where you check that the methods of this `Basic` class work as intended. Namely:

- `get_user_nodes`
- `get_attribute_nodes`
- `get_direct_user_nodes`
- `user_nodes`
- `remove_user_node`

Reply:
You are right, I will do this

Reply:
I have removed `user_nodes`

I am not sure how to test `remove_user_node`. It should be tested within `substitute` though. I originally named this function `_remove_user_node` as it shouldn't be called from outside `Basic` but codacy didn't let me keep that name and call the function for something other than `self`

Reply:
The tests are in the folder `tests/ast/`

Reply:
Perfect, thanks!